python/lib/database.py

```diff
@@ -0,0 +1,608 @@
+
+            # Add 1000 to the current value and set the sequence to the new value
+            new_value = last_value + 999
+            sql = f"SELECT setval('{sequence}', {new_value});"
+            sql_print(sql)
+            cursor.execute(sql)
+
+        db.commit()
+
+
+def populate_lookup_tables(db):
+    with open("seeds/road_types.csv", "r") as f:
+        reader = csv.reader(f)
+        next(reader)  # Skip the header row
+        with db.cursor() as cursor:
+            for row in reader:
+                sql = "INSERT INTO cris_lookup.road_types (upstream_id, description) VALUES (%s, %s) RETURNING id;"
+                cursor.execute(sql, (row[0], row[1]))
+                returned_id = cursor.fetchone()["id"]
+                sql_print((sql % (row[0], f"'{row[1]}'")) + f" --> {returned_id}")
+            db.commit()
+
+    with open("seeds/unit_types.csv", "r") as f:
+        reader = csv.reader(f)
+        next(reader)  # Skip the header row
+        with db.cursor() as cursor:
+            for row in reader:
+                sql = "INSERT INTO cris_lookup.unit_types (upstream_id, description) VALUES (%s, %s) RETURNING id;"
+                cursor.execute(sql, (row[0], row[1]))
+                returned_id = cursor.fetchone()["id"]
+                sql_print((sql % (row[0], f"'{row[1]}'")) + f" --> {returned_id}")
+            db.commit()
+
+
+def refresh_materialized_views(db):
+    with db.cursor() as cursor:
+        sql = """
+        SELECT matviewname
+        FROM pg_matviews
+        WHERE schemaname = 'public'
+        """
+        cursor.execute(sql)
+        matviews = [row["matviewname"] for row in cursor.fetchall()]
+
+        for matview in matviews:
+            sql = f"REFRESH MATERIALIZED VIEW public.{matview};"
+            sql_print(sql)
+            cursor.execute(sql)
+
+    db.commit()
+
+
+def create_fact_tables(db):
+    sql_commands = [
+        """CREATE TABLE cris_facts.crashes (
+            id SERIAL PRIMARY KEY,
+            crash_id INTEGER NOT NULL,
+            primary_address TEXT,
+            road_type_id INTEGER,
+            location GEOMETRY(Point, 4326)
+        );
+        """,
+        """CREATE TABLE cris_facts.units (
+            id SERIAL PRIMARY KEY,
+            unit_id INTEGER NOT NULL,
+            crash_id INTEGER NOT NULL,
+            unit_type_id INTEGER
+        );
+        """,
+        """CREATE TABLE visionzero_facts.crashes (
+            id SERIAL PRIMARY KEY,
+            cris_id INTEGER NOT NULL REFERENCES cris_facts.crashes(id) ON DELETE CASCADE,
+            crash_id INTEGER,
+            primary_address TEXT,
+            road_type_id INTEGER,
+            location GEOMETRY(Point, 4326)
+        );
+        """,
+        """CREATE TABLE visionzero_facts.units (
+            id SERIAL PRIMARY KEY,
+            cris_id INTEGER NOT NULL REFERENCES cris_facts.units(id) ON DELETE CASCADE,
+            unit_id INTEGER,
```

Comment:
still looking at this, but it threw me off to see `cris_id` here and `unit_id` containing the CRIS unit ID in the other unit tables. can you give me an example of what the `unit_id` would be here? I could be missing something.